src/include/detail/ivf/index.h

```diff
@@ -95,51 +95,54 @@ int ivf_index(
       0,
       centroid_read_temporal_policy);
   centroids.load();
-  auto parts = detail::flat::qv_partition(centroids, db, nthreads);
-  // debug_matrix(parts, "parts");
+  // Find the centroid that is closest to each input vector.
+  auto parts = detail::flat::qv_partition(centroids, input_vectors, nthreads);
   {
     scoped_timer _{"shuffling data"};
-    std::unordered_set<ids_type> deleted_ids_set(
+    std::unordered_set<IdsType> deleted_ids_set(
         deleted_ids.begin(), deleted_ids.end());
+    // The number of vectors assigned to each centroid.
     std::vector<size_t> degrees(centroids.num_cols());
-    std::vector<ids_type> indices(centroids.num_cols() + 1);
     if (deleted_ids.empty()) {
-      for (size_t i = 0; i < db.num_cols(); ++i) {
+      for (size_t i = 0; i < input_vectors.num_cols(); ++i) {
         auto j = parts[i];
         ++degrees[j];
       }
     } else {
-      for (size_t i = 0; i < db.num_cols(); ++i) {
-        if (auto iter = deleted_ids_set.find(external_ids[i]);
-            iter == deleted_ids_set.end()) {
+      for (size_t i = 0; i < input_vectors.num_cols(); ++i) {
+        if (deleted_ids_set.find(external_ids[i]) == deleted_ids_set.end()) {
           auto j = parts[i];
           ++degrees[j];
         }
       }
     }
+
+    // The starting index of each partition in the shuffled data.
+    std::vector<IdsType> indices(centroids.num_cols() + 1);
     indices[0] = 0;
     std::inclusive_scan(begin(degrees), end(degrees), begin(indices) + 1);
 
-    std::vector<size_t> check(indices.size());
-    std::copy(begin(indices), end(indices), begin(check));
+    // std::vector<size_t> check(indices.size());
+    // std::copy(begin(indices), end(indices), begin(check));
 
     // @todo Add systematic mechanism for debugging algorithms / dat structures
     // debug_matrix(degrees, "degrees");
     // debug_matrix(indices, "indices");
 
     // Some variables for debugging
     // @todo remove these once we are confident in the code
-    auto mis = std::max_element(begin(indices), end(indices));
-    auto a = std::distance(begin(indices), mis);
-    auto b = std::distance(mis, end(indices));
-    auto misx = *mis;
+    // auto mis = std::max_element(begin(indices), end(indices));
+    // auto a = std::distance(begin(indices), mis);
```

Comment:
Why don't we remove these entirely?

Reply:
Sure, done, just was leaving b/c the comments somewhat suggested there was future work we wanted to do here with it.